### ISSUE TITLE
making RelFinder work with a SPARQL endpoint.

### ISSUE THREAD

**Issue by bluejoe2008** [grapheco/InteractiveGraph#15]
I'd like to have a longer chat with you to find out more about what you're doing and to share with you my interests.  We might find some good synergies

Thanks for your help in getting started with your InteractiveGraph software, I think it will be very useful.  I have been able to build the minified javascript file now and will look at greeting some graph files on my own data.  After that, I am interested in having something like relfinder that can work with a SPARQL endpoint.

**Comment by bluejoe2008** [comment]
The InteractiveGraph supplies three applications(navigator, explorer, and relFinder). With the help of Interactive-RDF, the InteracitveGraph can show the RDF data stored in RDF file(only n3 are supported now) or Jena(a RDF database). Yes, as you see, the InteractiveGraph is a frontend software, and the InteractiveGraph-RDF(also InteractiveGraph-neo4j) is a backend software. The frontend software doesn't store or manage the data, it only query and render the data from backend softwares as the user need. If you take care about the logic of the software， you may find that the frontend are running some queries to get specific data from backend end. For example, the navigator retrive and show all the data, the explorer retrive and show connected data of the node you click on the screen, the relFinder find all the data between to nodes(or triples in RDF data), and show. How does it implement this function? Actually, it generates the query, send the query to the backend, the backend execute the queries, get the data, send the data back to frontend.

Let's back to our enthusiastic user, it seems he wants to write SPARQL queries by himself to query the data, and show it in the InteractiveGraph(relFinder). It's realistic, however maybe not a good idea in this software. We designed many interfaces to translate user actions into query languages which the backend can understand. So we didn't supply the function to support user write their own queries. However, if this function is highly needed, we may consider to implement it in th future.

**Comment by pwin** [comment]
No, I don't want to write a SPARQL query myself - I want the front end to create a sequence of queries to allow the system to discover links between two entities at a specified distance from each other in the graph (distance = the number of node-edge-node jumps)

So, relfinder does this in 2 ways - see the code https://github.com/VisualDataWeb/RelFinder/blob/master/src/connection/SPARQLQueryBuilder.as
and the papers referenced in http://www.visualdataweb.org/relfinder.php

I looked at the code in InteractiveGraph ( https://github.com/grapheco/InteractiveGraph-RDF/blob/master/src/main/scala/org/interactivegraph/server/jena/services.scala ) and I don't think that it takes the same approach.   Am I missing something?
